libjwt/jwt.c

```diff
@@ -15,6 +15,55 @@
 #include "base64.h"
 #include "config.h"
 
+static jwt_alloc_t pfn_alloc = malloc;
+static jwt_realloc_t pfn_realloc = realloc;
+static jwt_free_t pfn_free = free;
```

Comment:
I would much rather set these to `NULL`, and make the below functions do:

    void *jwt_malloc(size_t size)
    {
        if (pfn_malloc)
            return pfn_malloc(size);
        else
            return malloc(size);
    }

This way, we can pass `NULL` to the `jwt_set_alloc()` for ones where we want to use native. Also. I believe some implementations may have `malloc` and friends set as a macro, and we can't pass them as pointers like you have.

This also makes it easy to selectively override some allocators, or revert to native, easily.

Reply:
Done